File: CS440_Project1_Sp_26_Starter_Package/starter_java/PCBStateSim.java
/*
 * CS 440 – PCB Simulator Starter (Java)
 * TODO: Add your name(s) and BearID(s)
 */

import java.util.*;

enum State { NEW, READY, RUNNING, WAITING, TERMINATED }

class PCB {
    int pid;
    String name;
    State state;
    int priority;
    int pc = 0;
    int cpuTime = 0;
}

public class PCBStateSim {
    // TODO: process table (Map<String, PCB>)
    // TODO: READY queue
    // TODO: WAITING queue
    // TODO: RUNNING reference
    // TODO: BearID auto-STATUS interval

    public static void main(String[] args) {
        // TODO: parse args (trace file)
        // TODO: read file line by line
        // TODO: dispatch commands
    }

    // TODO: handlers for CREATE, DISPATCH, TICK, BLOCK, WAKE, EXIT, STATUS, KILL
}

File: CS440_Project1_Sp_26_Starter_Package/starter_c/pcb_state_sim.c
/*
 * CS 440 – PCB Simulator Starter (C)
 * TODO: Add your name(s) and BearID(s)
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef enum {
    NEW, READY, RUNNING, WAITING, TERMINATED
} State;

typedef struct PCB {
    int pid;
    char name[32];
    State state;
    int priority;
    int pc;
    int cpuTime;
    struct PCB *next;
} PCB;

// TODO: process table
// TODO: READY queue
// TODO: WAITING queue
// TODO: RUNNING pointer
// TODO: BearID auto-STATUS interval

int main(int argc, char *argv[]) {
    // TODO: parse args
    // TODO: read trace file
    // TODO: dispatch commands
    return 0;
}

// TODO: implement command handlers
